hack/getting_started.sh

```diff
@@ -0,0 +1,551 @@
+
+    sudo systemctl status docker | grep -q "active (running)"
+    if [ $? -ne 0 ]; then
+        echo "Enable docker failed"
+        exit 
+    fi
+    echo "Enable docker success"
+
+    # Make sure current user has permission for docker, do this if not Create the docker group.
+    docker ps 2>&1 | grep -q "connect: permission denied"
+    if [ $? -eq 0 ]; then
+
+        grep -q "docker" /etc/group 
+        if [ $? -ne 0 ]; then
+            runCmd "sudo groupadd docker" 0
+        else
+            echo "group 'docker' already exists"
+        fi
+        USER=`whoami`
+        # Add your user to the docker group.
+        runCmd "sudo usermod -aG docker ${USER}" 0 "Add ${USER} to docker group"
+        echo "You would need to log out and log back in so that your group membership is re-evaluated. Rerun this script after that."
+        exit
+    fi
+    echo "current user has permission for docker"
+}
+
+function installKind() {
+    cmdName="kind"
+     ## check  if denpency is installed before
+    isCmdInstalled "${cmdName}"
+
+    ## install denpency if it not installed
+    if [ $? -eq 0 ] ; then
+        echo "Installing ${cmdName}..."
+        curl -Lo ./kind https://kind.sigs.k8s.io/dl/v0.11.1/kind-linux-amd64 && sudo install kind /usr/local/bin/kind
+        ## check  if denpency is installed successfully
+        isCmdInstalled  "${cmdName}"
+        exitIfNot $? 0 "Installing ${cmdName} failed, exit..."
+    fi
+}
+
+function installClusterctl() {
+    cmdName="clusterctl"
+     ## check  if denpency is installed before
+    isCmdInstalled "${cmdName}"
+
+    ## install denpency if it not installed
+    if [ $? -eq 0 ] ; then
+        echo "Installing ${cmdName}..."
+        curl -L https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.1.1/clusterctl-linux-amd64 -o clusterctl && sudo install clusterctl /usr/local/bin/clusterctl
+        ## check  if denpency is installed successfully
+        isCmdInstalled  "${cmdName}"
+        exitIfNot $? 0 "Installing ${cmdName} failed, exit..."
+    fi
+}
+
+function intallDependencies(){
+    installDocker
+    enableDocker
+    installKind
+    installClusterctl
+    commonInstall jq "sudo apt install -y jq"
+    commonInstall kubectl "sudo snap install kubectl --classic"
+    commonInstall make "sudo apt install -y build-essential"
+    commonInstall go "sudo snap install go --classic"
+}
+
+function commonInstall(){
+    cmdName=$1
+    installCmd=$2
+
+    ## check  if denpency is installed before
+    isCmdInstalled "${cmdName}"
+
+    ## install denpency if it not installed
+    if [ $? -eq 0 ] ; then
+        runCmd "${installCmd}" 0 "Installing ${cmdName}..."
+
+        ## check  if denpency is installed successfully
+        isCmdInstalled  "${cmdName}"
+        exitIfNot $? 0 "Installing ${cmdName} failed, exit..."
+    fi
+}
+
+function createKindCluster(){
+    echo  "Creating kind cluster..."
+
+    msg=$(kind create cluster 2>&1)
```

Comment:
Let's name this cluster as `kind-byoh`. Users might have a cluster called `kind` already present

Reply:
We should first check if `kind-byoh` is present. If so, delete and create a new one. 
Running this script means we are starting a fresh cluster. Should not rely on previous runs / states
This will save us lot of effort in checking if byoh is installed, cluster conf is present etc.